core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java

```diff
@@ -7,8 +7,14 @@
  * have filters that respond specifically to goal queries, but currently we do not.
  */
 public class GoalQuery extends MatchQuery {
+  private final Goal<?> goal;
 
-  public GoalQuery(Goal goal) {
+  public GoalQuery(Goal<?> goal) {
```

Comment:
Would not `<? extends GoalDefinition>` be "more" correct?

Reply:
Goal itself is already bounded to goal definition, pretty sure that is enforced here anyways
Regardless it simply is a way not to raise a warning

Reply:
Oh yeah. Even though Intellisense says it returns "capture of ?" instead of "capture of ? extends GoalDefinition" it seems to work as expected. I guess it would only be for clarity when reading the source at this point.

Reply:
![image](https://user-images.githubusercontent.com/11789291/163359251-161ef940-2ea7-4677-8d58-d94aa52785d0.png)

Goal's generic already supports only stuff that extend GoalDefinition, so it's just wouldn't be allowed to be anything else regardless